build.gradle

```diff
@@ -44,14 +44,20 @@ repositories {
     mavenCentral()
 }
 
+configurations {
+    provided
+}
+
 dependencies {
     compile("com.emc.ecs:object-client:3.0.1") {
         exclude group: "org.slf4j", module: "slf4j-api"
         exclude group: "org.slf4j", module: "slf4j-log4j12"
     }
 
-    compileOnly "org.apache.spark:spark-core_$versions.scala:$versions.spark",
-                "org.apache.spark:spark-sql_$versions.scala:$versions.spark"
+    provided "org.apache.spark:spark-core_$versions.scala:$versions.spark",
+             "org.apache.spark:spark-sql_$versions.scala:$versions.spark"
+
+    compileOnly configurations.provided
```

Comment:
I don't see why you'd need a custom configuration called 'provided' when 'compileOnly' is exactly that.